File: app/src/main/java/org/schulcloud/mobile/data/model/Account.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmModel;

import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@RealmClass
public class Account implements RealmModel {
    @PrimaryKey
    public String _id;
    public String username;
    public String password;
    public String updatedAt;
    public String createdAt;
    public User userId;


    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public void setUsername(String username){this.username = username;}

    public String getUsername(){return this.username;}

    public void setPassword(String password){}

    public String getUpdatedAt(){return  this.updatedAt;}

    public void setUpdatedAt(String updatedAt){this.updatedAt = updatedAt;}

    public String getCreatedAt(){return this.createdAt;}

    public void setCreatedAt(String createdAt){this.createdAt = createdAt;}

    public User getUserId(){return this.userId;}

    public void setUserId(User userId){this.userId = userId;}

}

File: app/src/main/java/org/schulcloud/mobile/models/user/Account.kt
package org.schulcloud.mobile.models.user

import com.google.gson.annotations.SerializedName
import io.realm.RealmModel
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Account: RealmObject() {
    @PrimaryKey
    @SerializedName("_id")
    var id: String = ""

    var username: String? = null
    var password: String? = null
    var userId: String? = null

    var newPassword: String? = null
    var newPasswordRepeat: String? = null
}